server/routerlicious/packages/lambdas-driver/src/document-router/documentPartition.ts

```diff
@@ -88,8 +88,14 @@ export class DocumentPartition {
                 this.q.resume();
             },
             (error) => {
-                context.error(error, { restart: true, tenantId, documentId });
-                this.q.kill();
+                // There is no need to pass the message to be checkpointed to markAsCorrupt().
+                // The message, in this case, would be the head in the DocumentContext (the DocumentLambda
+                // that creates this DocumentPartition will also put the same message in the queue).
+                // So the DocumentPartition will see that message in the queue above, and checkpoint it
+                // since the document was marked as corrupted.
+                this.markAsCorrupt(error);
+                context.error(error, { restart: false, tenantId, documentId });
```

Comment:
This `context.error` line is not needed because `markAsCorrupt` will do the same thing